src/Microsoft.ML.PipelineInference/AutoInference.cs

```diff
@@ -536,7 +539,26 @@ public PipelinePattern[] GetNextCandidates(int numberOfCandidates)
                 var currentBatchSize = numberOfCandidates;
                 if (_terminator is IterationTerminator itr)
                     currentBatchSize = Math.Min(itr.RemainingIterations(_history), numberOfCandidates);
-                BatchCandidates = AutoMlEngine.GetNextCandidates(_sortedSampledElements.Select(kvp => kvp.Value), currentBatchSize);
+                BatchCandidates = AutoMlEngine.GetNextCandidates(
+                    _sortedSampledElements.Select(kvp => kvp.Value), 
+                    currentBatchSize,
+                    this._columnPurpose);
+
+                var h = _env.Register("AutoMlMlState");
```

Comment:
> var h = _env.Register("AutoMlMlState"); [](start = 15, length = 40)

you should have _host field, no reason to create new host variable.

Reply:
Thanks Ivan. I will take care of this in the next iteration